### PR TITLE
VBufStorage: Fix crash by not allowing the reusing of nodes with differing parents.

### LINKED ISSUE #8924: Crash in VBufStorage_buffer_t::deleteNode in Firefox Nightly with Gmail
### Steps to reproduce:

1. I was in a Gmail conversation.
2. Pressed Space on the More Actions for the conversation in the upper tool bar.
3. Arrowed down in the menu, wanted to add a filter. Got to the Add To Tasks item.
4. Press DownArrow once more.

### Actual behavior:

Reproducible crash that closed Firefox. Sometimes, Crash Reporter doesn't even come up. But once it did, and it gave me [this crash report](https://crash-stats.mozilla.com/report/index/6c6c3806-ad3e-4b1d-b625-37f720181106).

### Expected behavior:

No crash.

#### NVDA Installed/portable/running from source:

Installed.

#### NVDA version:

NVDA version alpha-16220,87eb36f8

#### Windows version:

Windows 10 insider 18272.

#### Name and version of other software in use when reproducing the issue:

Firefox 65.0a1 Nightly from November 6.

#### Other information about your system:

Gmail is in New Design, I believe this is the default now, and you cannot eve turn that off any more. It's the standard Gmail, not G Suite.

#### Does the issue still occur after restarting your PC?

Yes. Reproducible on my system.

#### Have you tried any other versions of NVDA?

Yes, 2018.3.x does not crash. I suspect this is a result of either the VBuf speedup or the merging of DLLs, or both.

CC @michaelDCurran @jcsteh.

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
Closes #8924 

### Summary of the issue:
A web author can move a node from one place to another in the DOM. A web author can also move an accessible from one place to another within the accessibility tree, by dynamically changing aria-owns.
With the merging of pr #8678 this can in some cases cause NVDA to crash the web browser if NVDA's virtualBuffer for the document tries to reusethe moved node, as two separate updates may try to touch the same thing.
Examples of this are in #8924 including a real-life testcase in Gmail, and a specific code fragment.

### Description of how this pull request fixes the issue:
NVDA's virtualBuffer code now refuses to reuse an existing node when re-rendering a parent if the existing node has a different parent to what is being re-rendered. 
However, it still allows nodes to change ordering within a given parent (if the existing node does not directly disallow this, such as in tables). If we do not keep this, the German wikipedia page becomes as slow as it used to be in NVDA 2018.3.
 
### Testing performed:
Tested the speed of the German wikipedia page. Tested the speed of moving up and down the inbox in Gmail.
Tested the code fragment in #8924.
However, waiting on @marcoZehe to report if the try build in #8924 successfully fixes the crash for him.
I was never able to reproduce the crash.
 
### Known issues with pull request:
None.

### Change log entry:
None needed.

### CODE REVIEW

File: nvdaHelper/vbufBase/backend.cpp [hunk 1]
```diff
@@ -272,6 +276,18 @@ VBufStorage_controlFieldNode_t* VBufBackend_t::reuseExistingNodeInRender(VBufSto
 		LOG_DEBUG(L"Existing node refuses to be reused");
 		return nullptr;
 	}
+	// We only allow reusing nodes that remain at the same position in the tree.
+	// I.e. We never reuse a node that is from  a different parent.
```
Comment (jcsteh) [inline nvdaHelper/vbufBase/backend.cpp]: Even though you mention parent here, the term "position" is used, which might make someone think this also applies to movement within the same parent. You could either just shorten this to talk about the same parent or clarify by saying that movement within the same direct parent is still okay.

**Review by jcsteh — APPROVED** [review]
> However, it still allows nodes to change ordering within a given parent (if the existing node does not directly disallow this, such as in tables). If we do not keep this, the German wikipedia page becomes as slow as it used to be in NVDA 2018.3.

When you say change ordering, I assume you mean this includes adding or removing another node? If so, changing that would probably defeat the whole point of the speed-ups, no? I guess we can't tell the difference at this point between a node being added between two existing nodes (e.g. acb) and two nodes switching places (e.g. ba)? (Where a, b and c are chronologically ordered.)

**Comment by michaelDCurran** [comment]
@marcoZehe has confirmed this fixes the crashes he was experiencing.
